feg/gateway/services/session_proxy/credit_control/model_conversion.go

```diff
@@ -45,6 +46,17 @@ func getCreditUnit(volume *uint64) *protos.CreditUnit {
 	return &protos.CreditUnit{IsValid: true, Volume: *volume}
 }
 
+func GetVirtualApnRule(apnFilter, apnOverwrite string) *VirtualApnRule {
+	reg, err := regexp.Compile(apnFilter)
+	if err != nil {
+		return nil
```

Comment:
If we're not returning an error out of this, we should log it. Or we could make this function attached to the `VirtualApnRule` structure and just called it `FromMconfig` or something like that. Then you can return an error/nil and have the caller handle the error.

Reply:
we should always avoid ignoring errors. This is the best way to hide a needle on a haystak

Reply:
Not sure to get your comment about ignoring errors. A log was printing an error if this function was returning nil
log.Printf("%s Managed Gx Virtual APN Rule Config Load Error", virtualApnlCfg.GetApnFilter())

Unless you mean we should abort the startup because of a wrong regex!

Reply:
The problem here is using nil to signal an error, which is a bad pattern. Either you handle the error in this function (printing the log) or you return the error explicitly and handle by the receiver.